### DISCUSSION TITLE
Add %06X support to MQTT Username field

### DISCUSSION THREAD

**Discussion by ajaypala** [arendst/Tasmota#20830] [Ideas]
Following good security practices that each device should use a unique username and password, it would make setup much easier if I could use %06X in the MQTT user name allowing the part of MAC address to form part of the username.

**Comment by s-hadinger** [comment]
Sorry but I don't quite agree. Having device names makes it much easier to manage in a home environment than Mac addresses. Additionally Mac address would require to reconfigure some significant configuration when changing a faulty device

**Reply by barbudor** [reply]
I'm setting my mqttuser the same as topic which is unique per device. Each device has it's own password

However the MQTT client is set to the full MAC address %12X

**Reply by sfromis** [reply]
As depending on the MAC makes it a mess to replace a failed device, I'm deliberately avoiding any usage of that in names, but provide my own unique names for device topic, hostname etc.

Those `%nnX` references are not intended for general usage, but *only* to allow a Tasmota device coming online first time, with a default naming to avoid clashes. Supplying my own names is the first thing I do after that.
